website/docs/development/controller-design.md

```diff
@@ -137,4 +137,23 @@ Deletion dependencies are handled by `dependency.DeletionGuardDependency`. Refer
 
 Methods on the deletion dependency will add the deletion guard to the manager, and therefore ensure that any added finalizers are removed at the correct time. However, the deletion guard is not responsible for adding finalizers to dependencies. This is the responsibility of the controller. `DeletionGuardDependency` provides the convenience method `GetDependencies` which will return all managed dependencies, ensuring the required finalizer has been added.
 
-Finalizers should be added at the last possible moment, namely immediately before their OpenStack resource is about to be referenced in an OpenStack create/update operation. The reason for this is, again, usability. If a user is provisioning a large number of resources and a failure happens somewhere, we want them to be able to delete, e.g. a failed Network. We don't want them to have to manually remove the Subnet finalizer from the Network for a Subnet which was never created because of the failure to provision its Network. Therefore we should not add a finalizer to Network until immediately before calling Create on the Subnet.
\ No newline at end of file
+Finalizers should be added at the last possible moment, namely immediately before their OpenStack resource is about to be referenced in an OpenStack create/update operation. The reason for this is, again, usability. If a user is provisioning a large number of resources and a failure happens somewhere, we want them to be able to delete, e.g. a failed Network. We don't want them to have to manually remove the Subnet finalizer from the Network for a Subnet which was never created because of the failure to provision its Network. Therefore we should not add a finalizer to Network until immediately before calling Create on the Subnet.
+
+## Logging
+We have 4 logging levels: Status, Info, Verbose and Debug.
```

Comment:
Use backticks for literal values. Additionally, it would be good to link to the file where these logging levels are defined:
```suggestion
We have [4 logging levels](https://github.com/k-orc/openstack-resource-controller/blob/main/internal/logging/levels.go): `Status`, `Info`, `Verbose` and `Debug`.
```